fails to build with libc++  by default

this is because:

- the default c++ std is c++17, which does not allow dynamic throw specifiers
- _GLIBCXX_THROW is used, which does not exist in libc++, and the fallback definition is just to `throw`

so, you get:  

```
../cupsfilters/pdftoraster.cxx:2204:34: error: ISO C++17 does not allow dynamic exception specifications [-Wdynamic-exception-spec]
void * operator new(size_t size) _GLIBCXX_THROW (std::bad_alloc)
                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../cupsfilters/pdftoraster.cxx:2201:24: note: expanded from macro '_GLIBCXX_THROW'
#define _GLIBCXX_THROW throw
```

in libstdc++ itself, _GLIBCXX_THROW expands to nothing for c++ >=11: https://github.com/gcc-mirror/gcc/blob/1e6815071fd137424f79ca79a6ba66b776a98778/libstdc%2B%2B-v3/include/bits/c%2B%2Bconfig#L219  

since the codebase already uses std=c++17, the best solution is to just remove the throw specifiers entirely

--- a/cupsfilters/pdftoraster.cxx
+++ b/cupsfilters/pdftoraster.cxx
@@ -2199,7 +2199,11 @@ cfFilterPDFToRaster(int inputfd,            // I - File descriptor input stream
 // For compatibility with g++ >= 4.7 compilers _GLIBCXX_THROW
 // should be used as a guard, otherwise use traditional definition
 #ifndef _GLIBCXX_THROW
-#define _GLIBCXX_THROW throw
+#  if __cplusplus >= 201103L
+#    define _GLIBCXX_THROW(_EXC)
+#  else
+#    define _GLIBCXX_THROW(_EXC) throw(_EXC)
+#  endif
 #endif
 
 void * operator new(size_t size) _GLIBCXX_THROW (std::bad_alloc)

